kernel/interrupts/src/x86_64/mod.rs

```diff
@@ -32,6 +32,19 @@ static RESERVED_IRQ_LIST: [u8; 3] = [
 ];
 
 
+#[macro_export]
+macro_rules! interrupt_handler {
+    ($name:ident, $x86_64_interrupt_number:expr, $stack_frame:ident, $code:block) => {
+        extern "x86-interrupt" fn $name(sf: $crate::InterruptStackFrame) {
+            let $stack_frame = &sf;
+            if let CallerMustSignalEoi = $code {
```

Comment:
this should probably be fully qualified in order to avoid name clashes or prevent calling crates from having to import it specifically.
```suggestion
            if let $crate::EoiBehavior::CallerMustSignalEoi = $code {
```